add spec for Array can be sliced with Enumerator::ArithmeticSequence

lxxxvi opened the pull request:
Hello 👋 

Today I'll try to cover this from #823 :

> (Array) Can be sliced with Enumerator::ArithmeticSequence

Thank you for your feedback.

eregon commented on core/array/shared/slice.rb:
```diff
@@ -533,6 +533,25 @@ def to.to_int() -2 end
+  ruby_version_is "3.0" do
+    it "can be sliced with Enumerator::ArithmeticSequence" do
+      a = [0, 1, 2, 3, 4, 5]
+      a.send(@method, eval("(0..).step(2)")).should == [0, 2, 4]
```
```suggestion
      a.send(@method, eval("(0..).step(1)")).should == [0, 1, 2, 3, 4, 5]
      a.send(@method, eval("(0..).step(2)")).should == [0, 2, 4]
```
to add a basic example with step 1

eregon reviewed (approved):
Thanks, looks great!

eregon commented on core/array/shared/slice.rb:
```diff
@@ -533,6 +533,25 @@ def to.to_int() -2 end
+      a.send(@method, eval("(2..-2).step(2)")).should == [2, 4]
+      a.send(@method, eval("(2...-2).step(2)")).should == [2]
+      a.send(@method, eval("(2..).step(-2)")).should == [2, 0]
+      a.send(@method, eval("(2..).step(-3)")).should == [2]
```
It would be useful to separate these examples into a separate `it` and use the description to describe what it does.
Could you do that?

lxxxvi replied:
@eregon Sure, will do!

eregon commented:
https://rubyreferences.github.io/rubychanges/3.0.html#array-slicing-with-enumeratorarithmeticsequence might be useful

eregon commented:
@zverok @mrkn Could you take a look at the cases marked with `???` and see if they make sense to you or if they are bugs?

lxxxvi commented:
Hi @eregon 

As you suggested, I moved the specs that were marked with `???` to [another branch](https://github.com/lxxxvi/spec/tree/array-can-be-sliced-with-arithmeticsequence-questionable-results-with-negative-steps) and removed them from this branch.

Letme know if you want any more changes, thank you!
